test/metabase/models/pulse_test.clj

```diff
@@ -252,6 +254,16 @@
                                                                          {:id (mt/user->id :crowberto)}]}]
                                         :skip_if_empty false}))))))
 
+(deftest dashboard-subscription-update-test
+  (testing "collection_id and dashboard_id of a dashboard subscription cannot be directly updated"
+      (mt/with-temp* [Collection [{collection-id :id}]
+                      Dashboard  [{dashboard-id :id}]
+                      Pulse      [pulse {:dashboard_id dashboard-id :collection_id collection-id}]]
+        (is (thrown? Exception
+              (db/update! Pulse (u/the-id pulse) {:collection_id (inc collection-id)})))
```

Comment:
I think it's usually better to do a `thrown-with-msg?` assertion because then we can actually confirm that the exception getting thrown is the one we expect. For all we know this could just be failing because of an NPE or something and the error we expected is not actually getting triggered